test/functional/kaui/admin_allowed_users_controller_test.rb

```diff
@@ -129,6 +153,30 @@ class AdminAllowedUsersControllerTest < Kaui::FunctionalTestHelper
       put :add_tenant, :allowed_user => allowed_user, :tenant_1 => nil
       assert_equal 'Successfully set tenants for user', flash[:notice]
       assert_response :redirect
+      # validate redirect path
+      assert_equal "/kaui/admin_allowed_users/#{au.id}", URI(@response.get_header('Location')).path
     end
+
+    private
+
+      def get_allowed_username(response_body)
+        fields = /<input.*type="text".*value="(?<value>.+)".*name=.allowed_user.kb_username...*>/.match(response_body)
```

Comment:
Similarly here, if we end up doing similar extractions in other places, would be worth extracting these into helpers (e.g. `get_value_from_input_field`).